### ISSUE TITLE
Problem with Thread ???

### ISSUE THREAD

**Issue by hnanhtuan** [soumith/imagenet-multiGPU.torch#102]
Hi,

I got the error below. Can you help to pinpoint the problem? Thank you very much.

qlua main.lua -cache /home/anhxtuan/Documents/Datasets/ILSVRC2010/trainCache -data /home/anhxtuan/Documents/Datasets/ILSVRC2010/ -nDonkeys 2 -nEpochs 1 -batchSize 20
-- ignore option cache
batchSize	20	128
-- ignore option optimState
nEpochs	1	55
-- ignore option netType
-- ignore option data
-- ignore option retrain
=> Creating model from file: models/alexnetowtbn.lua
=> Model
nn.Sequential {
  [input -> (1) -> (2) -> output]
  (1): nn.Sequential {
    [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> (11) -> (12) -> (13) -> (14) -> (15) -> (16) -> (17) -> (18) -> output]
    (1): cudnn.SpatialConvolution(3 -> 64, 11x11, 4,4, 2,2)
    (2): cudnn.SpatialBatchNormalization
    (3): cudnn.ReLU
    (4): cudnn.SpatialMaxPooling(3x3, 2,2)
    (5): cudnn.SpatialConvolution(64 -> 192, 5x5, 1,1, 2,2)
    (6): cudnn.SpatialBatchNormalization
    (7): cudnn.ReLU
    (8): cudnn.SpatialMaxPooling(3x3, 2,2)
    (9): cudnn.SpatialConvolution(192 -> 384, 3x3, 1,1, 1,1)
    (10): cudnn.SpatialBatchNormalization
    (11): cudnn.ReLU
    (12): cudnn.SpatialConvolution(384 -> 256, 3x3, 1,1, 1,1)
    (13): cudnn.SpatialBatchNormalization
    (14): cudnn.ReLU
    (15): cudnn.SpatialConvolution(256 -> 256, 3x3, 1,1, 1,1)
    (16): cudnn.SpatialBatchNormalization
    (17): cudnn.ReLU
    (18): cudnn.SpatialMaxPooling(3x3, 2,2)
  }
  (2): nn.Sequential {
    [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> (11) -> output]
    (1): nn.View(9216)
    (2): nn.Dropout(0.500000)
    (3): nn.Linear(9216 -> 4096)
    (4): cudnn.BatchNormalization
    (5): cudnn.ReLU
    (6): nn.Dropout(0.500000)
    (7): nn.Linear(4096 -> 4096)
    (8): cudnn.BatchNormalization
    (9): cudnn.ReLU
    (10): nn.Linear(4096 -> 1000)
    (11): cudnn.LogSoftMax
  }
}
=> Criterion
nn.ClassNLLCriterion
==> Converting model to CUDA
table: 0x40585038
Saving everything to: /home/anhxtuan/Documents/Datasets/ILSVRC2010/trainCache/alexnetowtbn,batchSize=20,nEpochs=1/Thu01Jun201709:11:43AM
Starting donkey with id: 1 seed: 3
Starting donkey with id: 2 seed: 4
Loading train metadata from cache
Loading train metadata from cache
Loading test metadata from cache
Loading test metadata from cache
qlua: ...an/tools/torch/install/share/lua/5.1/threads/threads.lua:183: [thread 1 callback] ...nhxtuan/tools/torch/install/share/lua/5.1/torch/File.lua:375: unknown object
stack traceback:
	[C]: in function 'error'
	...nhxtuan/tools/torch/install/share/lua/5.1/torch/File.lua:375: in function 'readObject'
	...nhxtuan/tools/torch/install/share/lua/5.1/torch/File.lua:368: in function 'readObject'
	...nhxtuan/tools/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
	...nhxtuan/tools/torch/install/share/lua/5.1/torch/File.lua:353: in function 'readObject'
	...nhxtuan/tools/torch/install/share/lua/5.1/torch/File.lua:409: in function 'load'
	...GitHub/Working/imagenet-multiGPU.torch-master/donkey.lua:133: in main chunk
	[C]: in function 'dofile'
	...5-GitHub/Working/imagenet-multiGPU.torch-master/data.lua:30: in function <...5-GitHub/Working/imagenet-multiGPU.torch-master/data.lua:24>
	[C]: in function 'xpcall'
	...an/tools/torch/install/share/lua/5.1/threads/threads.lua:234: in function 'callback'
	...tuan/tools/torch/install/share/lua/5.1/threads/queue.lua:65: in function <...tuan/tools/torch/install/share/lua/5.1/threads/queue.lua:41>
	[C]: in function 'pcall'
	...tuan/tools/torch/install/share/lua/5.1/threads/queue.lua:40: in function 'dojob'
	[string "  local Queue = require 'threads.queue'..."]:13: in main chunk
stack traceback:
	[C]: at 0x7f812da39f50
	[C]: in function 'error'
	...an/tools/torch/install/share/lua/5.1/threads/threads.lua:183: in function 'dojob'
	...an/tools/torch/install/share/lua/5.1/threads/threads.lua:264: in function 'synchronize'
	...an/tools/torch/install/share/lua/5.1/threads/threads.lua:142: in function 'specific'
	...an/tools/torch/install/share/lua/5.1/threads/threads.lua:125: in function 'Threads'
	...5-GitHub/Working/imagenet-multiGPU.torch-master/data.lua:19: in main chunk
	[C]: in function 'dofile'
	main.lua:37: in main chunk

And here is my luarocks installed list:

> Installed rocks:
----------------
argcheck
cudnn
cunn
cutorch
cwrap
dok
env
gnuplot
graph
image
lmdb.torch
loadcaffe
lua-cjson
luaffi
luafilesystem
matio
moses
nn
nngraph
nninit
nnx
optim
paths
penlight
qtlua
qttorch
sundown
sys
threads
torch
trepl
xlua
